man/faq/developer/howto-coercion-data-frame.Rmd

```diff
@@ -141,47 +154,103 @@ knitr_local_registration("base::print", "my_tibble")
 This subclass is very simple. All it does is modify the header.
 
 ```{r}
-df <- my_tibble(x = 1, y = 1:2)
+red <- my_tibble("red", x = 1, y = 1:2)
+red
 
-df
+red[2]
 
-df[2]
+green <- my_tibble("green", z = TRUE)
+green
 ```
 
 Combinations do not work properly out of the box, instead vctrs falls back to a bare tibble:
 
 ```{r}
-vec_rbind(df, tibble::tibble(x = 10:12))
+vec_rbind(red, tibble::tibble(x = 10:12))
 ```
 
-Instead of falling back to a data frame, we would like to return a `<my_tibble>` when combined with a data frame or a tibble. This subclass of tibble is a supertype of tibble and data frame, i.e. it is the richer type.
+Instead of falling back to a data frame, we would like to return a `<my_tibble>` when combined with a data frame or a tibble. Because this subclass has more metadata than normal data frames (it has a colour), it is a _supertype_ of tibble and data frame, i.e. it is the richer type. This is similar to how a grouped tibble is a more general type than a tibble or a data frame. Conceptually, the latter are pinned to a single constant group.
+
+The coercion methods for data frames operate in two steps:
+
+- They check for compatible subclass attributes. In our case the tibble colour has to be the same, or be undefined.
```

Comment:
Wouldn't we usually take the colour from the lhs? (like the tz?)

Reply:
In this case this is a data attribute, not a detail attribute, so the colour has to match. I should add a discussion about this.